lib/new_relic/agent/instrumentation/action_mailer_subscriber.rb

```diff
@@ -0,0 +1,66 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/notifications_subscriber'
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      class ActionMailerSubscriber < NotificationsSubscriber
+        def start(name, id, payload)
+          return unless state.is_execution_traced?
+
+          start_segment(name, id, payload)
+        rescue => e
+          log_notification_error(e, name, 'start')
+        end
+
+        def finish(name, id, payload)
+          return unless state.is_execution_traced?
+
+          finish_segment(id, payload)
+        rescue => e
+          log_notification_error(e, name, 'finish')
+        end
+
+        def start_segment(name, id, payload)
+          segment = Tracer.start_segment(name: metric_name(name, payload))
+          segment.params[:key] = payload[:key]
+          segment.params[:exist] = payload[:exist] if payload.key?(:exist)
+          push_segment(id, segment)
+        end
+
+        def finish_segment(id, payload)
+          if segment = pop_segment(id)
+            if exception = exception_object(payload)
+              segment.notice_error(exception)
+            end
+            segment.finish
+          end
+        end
+
+        def metric_name(name, payload)
+          service = payload[:service]
+          method = method_from_name(name)
+          "Ruby/ActionMailer/#{service}Service/#{method}"
+        end
+
+        PATTERN = /\Aservice_([^\.]*)\.action_mailer\z/
```

Comment:
This may need to be changed too. I don't believe all the `action_mailer` events are prefixed with `service`.

Reply:
Great call

Reply:
Reworked with [354f0de](https://github.com/newrelic/newrelic-ruby-agent/pull/1740/commits/354f0de483791f52daec5bae941b6e513015e127)